recipes/z3/all/conanfile.py

```diff
@@ -48,9 +51,16 @@ def config_options(self):
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
+        if self.options.multiprecision == "internal":
+            self.provides.append("gmp")
 
     def requirements(self):
-        self.requires("mpir/3.0.0")
+        if self.options.multiprecision == "mpir":
+            self.requires("mpir/3.0.0")
+        elif self.options.multiprecision == "gmp":
+            self.requires("gmp/6.2.1")
+        elif self.options.multiprecision == "internal":
+            pass
```

Comment:
```suggestion
            self.output.warn(f"{self.name} will build the internal implementation of multiprecision.")
```
At least we will have some hint about internal behavior.

Reply:
Ok, but not a warning. There is no issue to warn about, it's a feature.

Reply:
I would just unconditionally output it with `self.output.info`.

Reply:
Done.